src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -189,8 +189,20 @@ number :: Parser (Either Scientific Integer)
 number = do
   (numText, num) <- match (tok scientific)
   pure $ case Data.Text.find (== '.') numText of
+      -- Number specified with decimals, so store as a 'Scientific'
```

Comment:
From the spec: http://spec.graphql.org/June2018/#sec-Int-Value
> An Int number is specified without a decimal point or exponent (ex. 1).

http://spec.graphql.org/June2018/#sec-Float-Value

> A Float number includes either a decimal point (ex. 1.0) or an exponent (ex. 1e50) or both (ex. 6.0221413e23).

Reply:
So we just need to change `Data.Text.find (== '.') numText` to accommodate for the presence of exponent? Or is that readily available by some method on `Scientific`?

Reply:
Excellent, so the spec essentially tells us to follow the alternative scheme I described in my comment above.

Yes, I suppose the easiest is to change the check you mentioned to accommodate for this. In fact, `Scientific` can't make this distinction, as it saves even full integers such as `34325` in scientific notation, as `34325E0` (loosely speaking).